tox.ini

```diff
@@ -4,12 +4,67 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py37, py38, py39, py310
+skip_missing_interpreters = true
+envlist =
+    lint
+    python-azdev-unit
 
-[testenv]
+[base]
 deps =
+    -r dev_requirements
+description =
+# py{x} determines python version for environment.
+#    A valid interpreter must be installed (and added to PATH)
+#    for each selected version, otherwise the run will be skipped
+# az{x} determines:
+#    {min} - minimum supported azure-cli
+#    {cur} - latest released azure-cli
+#    {dev} - installs azure-cli from ../azure-cli
+# int/unit determines test suites to run
+# list all available tox environments with: `tox -av`
+    python: Local Python
+    py3.7: Python 3.7
+    py3.8: Python 3.8
+    py3.9: Python 3.9
+    py3.10: Python 3.10
+    azmin: min azure-cli
+    azcur: current azure-cli
+    azdev: dev azure-cli
+    int: Integration Tests
+    unit: Unit Tests
+
+[testenv:lint]
+description = run linter
+deps =
+    {[base]deps}
     azure-cli
 commands =
-    pip install -r dev_requirements
-    pip install --upgrade ../azure-cli/src/azure-cli-testsdk
-    pytest azext_iot/tests
+    flake8 azext_iot/
+    pylint azext_iot/ --rcfile=.pylintrc
+
```

Comment:
What are your thoughts about incorporating the `azdev` linter? Specifically thinking of the help/ref docs linting.

Reply:
I think it's definitely the direction to head.
I have some more small changes in this PR to at least get the initial pipeline / task squared away (python binary location, flake arguments, etc) - I think this change will slot in nicely once I begin some secondary changes to match feature parity of the current merge pipeline.

Reply:
This will be part of a next PR that puts tox testing in the merge pipeline.